content/src/main/content/jcr_root/apps/acs-commons/touchui-widgets/source/touchui-composite-multifield.js

```diff
@@ -0,0 +1,303 @@
+
+        if (!name) {
+            return;
+        }
+
+        //strip ./
+        if (name.indexOf("./") === 0) {
+            name = name.substring(2);
+        }
+
+        record[name] = $field.val();
+
+        //remove the field, so that individual values are not POSTed
+        $field.remove();
+    }
+
+    //for getting the nested multifield data as js objects
+    function getRecordFromMultiField($multifield){
+        var $fieldSets = $multifield.find("[class='coral-Form-fieldset']"),
+            records = [], record, $fields, name;
+
+        $fieldSets.each(function (i, fieldSet) {
+            $fields = $(fieldSet).find("[name]");
+
+            record = {};
+
+            $fields.each(function (j, field) {
+                fillValue($(field), record);
+            });
+
+            if(!$.isEmptyObject(record)){
+                records.push(record);
+            }
+        });
+
+        return records;
+    }
+
+    function submitAction() {
+        var $form = $(this).closest("form.foundation-form"),
+            mName = $("[" + DATA_ACS_AEM_NESTED + "]").data("name"),
+            $fieldSets = $("[" + DATA_ACS_AEM_NESTED + "][class='coral-Form-fieldset']"),
+            record, $fields, $field, name, $nestedMultiField;
+
+        $fieldSets.each(function (i, fieldSet) {
+            $fields = $(fieldSet).children().children(CFFW);
+
+            record = {};
+
+            $fields.each(function (j, field) {
+                $field = $(field);
+
+                //may be a nested multifield
+                $nestedMultiField = $field.find("[data-init='multifield']");
+
+                if($nestedMultiField.length === 0){
+                    fillValue($field.find("[name]"), record);
+                }else{
+                    name = $nestedMultiField.find("[class='coral-Form-fieldset']").data("name");
+
+                    if(!name){
+                        return;
+                    }
+
+                    //strip ./
+                    name = name.substring(2);
+
+                    record[name] = getRecordFromMultiField($nestedMultiField);
+                }
+            });
+
+            if ($.isEmptyObject(record)) {
+                return;
+            }
+
+            //add the record JSON in a hidden field as string
+            $('<input />').attr('type', 'hidden')
+                .attr('name', mName)
+                .attr('value', JSON.stringify(record))
+                .appendTo($form);
+        });
+    }
+
+    //collect data from widgets in multifield and POST them to CRX as JSON
+    function collectDataFromFields(){
+        $(document).on("click", ".cq-dialog-submit", submitAction);
+    }
+
+    $(document).ready(function () {
+        addDataInFields();
+        collectDataFromFields();
+    });
+
+    //extend otb multifield for adjusting event propagation when there are nested multifields
+    //for working around the nested multifield add and reorder
+    CUI.Multifield = new Class({
```

Comment:
please do not attach that to the CUI namespace. You will not be happy about that in future ;)